omod/src/main/java/org/openmrs/module/outgoingmessageexceptions/web/controller/rest/OutgoingMessageExceptionsRestController.java

```diff
@@ -65,4 +74,58 @@ public String getMessageById(@PathVariable Integer id) throws JsonProcessingExce
 		logger.debug("Get Single message reached by message id");
 		return outgoingMessageExceptionsService.getMessageById(id);
 	}
+	
+	@RequestMapping(value = "/messages/{id}/retry", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity retryMessage(@PathVariable Integer id, @RequestBody(required = false) String retryRequestJson) {
+		logger.debug("Set a message with " + id + " id as retried");
+		try {
+			LocalDate timestamp = null;
+			RetryRequest retryRequest = null;
```

Comment:
We can just use RetryRequest as the request body param, you do not need to parse it fro json manually